apps/namadillo/src/App/Common/GasFeeModal.tsx

```diff
@@ -0,0 +1,102 @@
+import { Modal, SkeletonLoading } from "@namada/components";
+import { chainAssetsMapAtom } from "atoms/chain";
+import {
+  gasPriceForAllTokensAtom,
+  storageGasTokenAtom,
+} from "atoms/fees/atoms";
+import { tokenPricesFamily } from "atoms/prices/atoms";
+import BigNumber from "bignumber.js";
+import { useAtomValue, useSetAtom } from "jotai";
+import { IoClose } from "react-icons/io5";
+import { twMerge } from "tailwind-merge";
+import { GasConfig } from "types";
+import { toDisplayAmount } from "utils";
+import { unknownAsset } from "utils/assets";
+import { FiatCurrency } from "./FiatCurrency";
+import { TokenCurrency } from "./TokenCurrency";
+
+export const GasFeeModal = ({
+  gasConfig,
+  onClose,
+}: {
+  gasConfig: GasConfig;
+  onClose: () => void;
+}): JSX.Element => {
+  const setStorageGasToken = useSetAtom(storageGasTokenAtom);
+  const gasPriceForAllTokens = useAtomValue(gasPriceForAllTokensAtom);
+  const chainAssetsMap = useAtomValue(chainAssetsMapAtom);
+
+  const data = gasPriceForAllTokens.data ?? [];
+
+  const tokenAddresses = data.map((item) => item.token);
+  const gasDollarMap = useAtomValue(tokenPricesFamily(tokenAddresses));
+
+  return (
+    <Modal onClose={onClose}>
+      <div
+        className={twMerge(
+          "fixed min-w-[550px] top-1/2 left-1/2 -translate-y-1/2 -translate-x-1/2",
+          "px-6 py-7 bg-rblack border border-neutral-500 rounded-md"
+        )}
+      >
+        <i
+          className={twMerge(
+            "cursor-pointer text-white absolute right-3 top-3 text-xl",
+            "hover:text-yellow transition-colors"
+          )}
+          onClick={onClose}
+        >
+          <IoClose />
+        </i>
+        <div className="text-center">
+          <h2 className="font-medium">Select Gas Token</h2>
+          <div className="text-sm mt-1">
+            Gas fees deducted from your Namada accounts
+          </div>
+        </div>
+        <div className="flex flex-col mt-4 max-h-[60vh] overflow-auto">
+          {!data.length ?
+            <SkeletonLoading height="100px" width="100%" />
+          : data.map(({ token, minDenomAmount }) => {
```

Comment:
Maybe we should sort this list in order to display NAM token on top, what do you think?

Reply:
For sure! I'll do it